components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java

```diff
@@ -197,7 +197,7 @@ public boolean validateScopes(String username, Swagger swagger, MessageContext s
             org.apache.axis2.context.MessageContext axis2MessageContext =
                     ((Axis2MessageContext) synCtx).getAxis2MessageContext();
             String httpMethod = (String) axis2MessageContext.getProperty(APIConstants.DigestAuthConstants.HTTP_METHOD);
-            String resourceKey = apiElectedResource + ":" + httpMethod;
+            String resourceKey = apiContext + ":" + apiVersion + ":" + apiElectedResource + ":" + httpMethod;
```

Comment:
Better if we can use a constant for ":".

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/9250